packages/hardhat/contracts/DisperseFunds.sol

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+
+interface IERC20 is Ownable {
+    function transfer(address, uint256) external returns(bool);
+}
+
+contract DisperseFunds {
+    error UserClaimed();
+    address public saltAddr;
+    uint256 public constant saltFaucetAmount = 25 ether;
+    uint256 public constant daiFaucentAmount = 0.1 ether;
+    mapping (address => bool) addressClaimed;
+    
+
+    constructor(address _saltAddr) {
+        saltAddr = _saltAddr;
+    }
+
+    function disperseBatch(address[] users) external onlyOwner {
+        uint256 userLen = users.length;
+
```

Comment:
Maybe you can check if the contract balance and salt balance are enough to send to userLen users.